sqlglot/dialects/tsql.py

```diff
@@ -708,6 +716,53 @@ def parse_range():
 
             return partition
 
+        def _parse_declare(self) -> exp.Declare | exp.Command:
+            index = self._index
+            try:
+                expressions = []
+                while True:
+                    if self._match(TokenType.PARAMETER, advance=False):
+                        var = self._parse_id_var()
+                        size = None
+                        value = None
+                        # moving past the AS keyword if it's there
+                        self._match_text_seq("AS")
+
+                        if self._match(TokenType.TABLE):
+                            table = self.expression(exp.Table, this=var)
+                            data_type = self._parse_schema(this=table)
```

Comment:
This also looks hacky, we should figure out a better way to construct the AST here.

Reply:
I'm not sure how to improve this. Is the `if` check the hacky part, or is it the fact that I'm creating the table expression manually and passing it into `_parse_schema`? The `if` check seems necessary to me and I'm not sure how else to get `_parse_schema` the table information any other way